bwapi/Library/BWAPILIB/Source/Types/UnitType.cpp

```diff
@@ -554,6 +554,28 @@ namespace BWAPI
     };
   }
 
+  namespace attackUpgrades
+  {
+   using namespace UpgradeTypes;
+   static const UpgradeType attackUpgrade[UnitTypes::Enum::MAX] = {

```

Comment:
What I do to check these is copy it in a text editor (sublime). Remove all spaces/newlines (regex replace `\s` with nothing), then replace commas with newlines. Then copy paste it into a spreadsheet with all the unit names in a separate column.

Some of the speed upgrades were still in this list, also infested kerrigan should use Zerg Melee Attacks. Also I had to double check but Scourge and Infested Terran don't have attack upgrades. The hero high templars do have normal ground attacks. Some other issues, here's the new list:

```cpp
Terran_Infantry_Weapons,Terran_Infantry_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,None,
Terran_Ship_Weapons,None,Terran_Infantry_Weapons,None,Terran_Ship_Weapons,None,None,None,
Terran_Infantry_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Infantry_Weapons,Terran_Ship_Weapons,None,Terran_Vehicle_Weapons,
Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,Terran_Ship_Weapons,Terran_Ship_Weapons,Terran_Ship_Weapons,Terran_Vehicle_Weapons,Terran_Vehicle_Weapons,
Terran_Infantry_Weapons,None,None,None,None,Zerg_Melee_Attacks,Zerg_Missile_Attacks,Zerg_Melee_Attacks,
Zerg_Melee_Attacks,None,None,Zerg_Flyer_Attacks,Zerg_Flyer_Attacks,None,None,,Zerg_Melee_Attacks,None,None,Zerg_Melee_Attacks,None,Zerg_Missile_Attacks,Zerg_Melee_Attacks,Zerg_Flyer_Attacks,
Zerg_Flyer_Attacks,None,Terran_Ship_Weapons,None,Protoss_Air_Weapons,Protoss_Ground_Weapons,Zerg_Flyer_Attacks,None,
None,Protoss_Ground_Weapons,Protoss_Ground_Weapons,None,Protoss_Ground_Weapons,None,Protoss_Air_Weapons,Protoss_Air_Weapons,
Protoss_Air_Weapons,Protoss_Air_Weapons,Protoss_Ground_Weapons,Protoss_Ground_Weapons,Protoss_Ground_Weapons,Protoss_Ground_Weapons,Protoss_Ground_Weapons,Protoss_Ground_Weapons,
Protoss_Air_Weapons,Scarab_Damage,Protoss_Air_Weapons,Scarab_Damage,None,Scarab_Damage,Protoss_Air_Weapons,Protoss_Ground_Weapons,
Protoss_Air_Weapons,None,None,None,None,None,None,None,
None,None,Protoss_Air_Weapons,Terran_Infantry_Weapons,Terran_Infantry_Weapons,None,Terran_Ship_Weapons,Zerg_Missile_Attacks,
Terran_Infantry_Weapons,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,None,None,None,None,None,None,None,
None,Unknown
```